src/Tools/AI Test Toolkit/src/Logs/AITRunHistory.Page.al

```diff
@@ -0,0 +1,222 @@
+                {
+                    Caption = 'Apply Filter';
+                    ToolTip = 'Specifies whether to filter results to a specific line or see results of the suite as a whole.';
+
+                    trigger OnValidate()
+                    begin
+                        UpdateRunHistory();
+                    end;
+                }
+            }
+
+            group("Line Filter")
+            {
+                Caption = 'Test Line Filter';
+                Visible = ApplyLineFilter;
+
+                field("Line No."; LineNoFilter)
+                {
+                    Caption = 'Test Line';
+                    ToolTip = 'Specifies the line to filter to.';
+                    Editable = false;
+
+                    trigger OnAssistEdit()
+                    begin
+                        LookupTestMethodLine();
+                    end;
+                }
+            }
+
+            group(History)
+            {
+                repeater("Run History")
+                {
+                    field(Version; Rec.Version)
+                    {
+                        Visible = ViewBy = ViewBy::Version;
+                        Caption = 'Version';
+                        ToolTip = 'Specifies the version.';
+                    }
+                    field(Tag; Rec.Tag)
+                    {
+                        Caption = 'Tag';
+                        ToolTip = 'Specifies the tag of the version.';
+                    }
+                    field("No. of Tests - By Version"; Rec."No. of Tests Executed")
+                    {
+                        Visible = ViewBy = ViewBy::Version;
+                        Caption = 'No. of Tests';
+                        ToolTip = 'Specifies the number of tests in the version.';
+                    }
+                    field("No. of Tests Passed - By Version"; Rec."No. of Tests Passed")
+                    {
+                        Visible = ViewBy = ViewBy::Version;
+                        Caption = 'No. of Tests Passed';
+                        ToolTip = 'Specifies the number of tests passed in the version.';
+                        Style = Favorable;
+                    }
+                    field("No. of Tests Failed - By Version"; Rec."No. of Tests Executed" - Rec."No. of Tests Passed")
+                    {
+                        Visible = ViewBy = ViewBy::Version;
+                        Editable = false;
+                        Caption = 'No. of Tests Failed';
+                        ToolTip = 'Specifies the number of tests that failed in the version.';
+                        Style = Unfavorable;
+
+                        trigger OnDrillDown()
+                        var
+                            AITLogEntryCU: Codeunit "AIT Log Entry";
+                        begin
+                            AITLogEntryCU.DrillDownFailedAITLogEntries(Rec."Test Suite Code", Rec."Line No. Filter", Rec.Version);
```

Comment:
```suggestion
                            Rec.SetRange(Version, Rec.Version)
                            AITLogEntryCU.DrillDownFailedAITLogEntries(Rec."Test Suite Code", Rec."Line No. Filter");
```

Reply:
This is where this function was used. Why not just set range here.

Reply:
Resolving as I think it's worth abstracting this in it's own overloaded function.